cupyx/scipy/signal/_polyutils.py

```diff
@@ -15,6 +15,31 @@ def roots(arr):
     return cupy.asarray(np.roots(arr))
 
 
+def poly(A):
+    """np.poly replacement for 2D A. Otherwise, use cupy.poly."""
+    sh = A.shape
+    if not (len(sh) == 2 and sh[0] == sh[1] and sh[0] != 0):
+        raise ValueError("input must be a non-empty square 2d array.")
+
+    import numpy as np
+
+    seq_of_zeros = np.linalg.eigvals(A.get())
+    seq_of_zeros = cupy.asarray(seq_of_zeros)
+
+    dt = seq_of_zeros.dtype
+    a = cupy.ones((1,), dtype=dt)
+    for zero in seq_of_zeros:
+        a = cupy.convolve(a, cupy.r_[1, -zero], mode='full')
+
+    if issubclass(a.dtype.type, cupy.complexfloating):
+        # if complex roots are all complex conjugates, the roots are real.
+        roots = cupy.asarray(seq_of_zeros, dtype=complex)
+        if cupy.all(cupy.sort(roots) == cupy.sort(roots.conjugate())):
+            a = a.real.copy()
```

Comment:
How about computing on CPU not only `eigvals` but also the whole of these calculations, then `cupy.asarray` just before return?